src/main/java/frc/utils/digitalinput/channeled/ChanneledDigitalInput.java

```diff
@@ -0,0 +1,31 @@
+package frc.utils.digitalinput.channeled;
+
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.wpilibj.DigitalInput;
+import frc.utils.digitalinput.DigitalInputInputsAutoLogged;
+import frc.utils.digitalinput.IDigitalInput;
+
+public class ChanneledDigitalInput implements IDigitalInput {
+
+	private final DigitalInput digitalInput;
+	private final Debouncer debouncer;
+	private final boolean inverted;
+
+	public ChanneledDigitalInput(int channel, double debounceTime, boolean inverted) {
+		this.digitalInput = new DigitalInput(channel);
+		this.debouncer = new Debouncer(debounceTime);
+		this.inverted = inverted;
+	}
+
+	@Override
+	public void updateInputs(DigitalInputInputsAutoLogged inputs) {
+		if (inverted) {
+			inputs.debouncedValue = !debouncer.calculate(digitalInput.get());
+			inputs.nonDebouncedValue = !digitalInput.get();
+		} else {
+			inputs.debouncedValue = debouncer.calculate(digitalInput.get());
+			inputs.nonDebouncedValue = digitalInput.get();
+		}
```

Comment:
use ternary for nonDebouncedValue 
use nonDebouncedValue  for debouncedValue

Reply:
what?

Reply:
thats like a lot of duplicate code here, use ternary (search what is it) for applying the nonDebouncedValue ,
and then the boolean doesnt matter if the debouncedValue if u use the inputs.nonDebouncedValue